app/Services/SprintChartService.php

```diff
@@ -50,19 +49,16 @@ private function getSprintById(int $sprintId): Sprint
             $sprint = $this->buildNewSprintFromJiraResponse($sprintResponse, $issuesResponse);
             $this->sprintRepository->save($sprint);
         }
-        if (Carbon::now()->isWeekday()) {
-            
-            $period = CarbonPeriod::create($sprint->getStartDate(), $sprint->getEndDate());
 
-            if ($period->contains(Carbon::now())) {
+        if (Carbon::now()->isWeekday()) {
+            if ($sprint->containsToday()) {
                 $issuesResponse = $this->getIssuesFromJiraBySprintId($sprintId);
                 $day = $this->buildNewDayFromJiraResponse($issuesResponse, $sprintId);
                 $this->dayRepository->save($day);
+                $sprint = $this->sprintRepository->loadById($sprintId);
             }
         }
```

Comment:
You could combine these and decrease the indent level
```suggestion
        if (Carbon::now()->isWeekday() && $sprint->containsToday()) {
            $issuesResponse = $this->getIssuesFromJiraBySprintId($sprintId);
            $day = $this->buildNewDayFromJiraResponse($issuesResponse, $sprintId);
            $this->dayRepository->save($day);
            $sprint = $this->sprintRepository->loadById($sprintId);
        }
```
OR if you want to get real fancy you could make a function on `Sprint` called `shouldUpdateToday` like
```php
public function shouldUpdateToday(): bool
{
    return Carbon::now()->isWeekday() && $this->containsToday();
}
```
and then change the if in the SprintChartService to be
```php
        if ($sprint->shouldUpdateToday()) {
            $issuesResponse = $this->getIssuesFromJiraBySprintId($sprintId);
            $day = $this->buildNewDayFromJiraResponse($issuesResponse, $sprintId);
            $this->dayRepository->save($day);
            $sprint = $this->sprintRepository->loadById($sprintId);
        }
```

Reply:
As we discussed, the weekend isn't part of the actual sprint so I've followed the second suggestion but reconciled the changes to just the `containsToday()` method. I did so by adding a guard:
```php
    public function containsToday(): bool
    {
        if (Carbon::now()->isWeekend()) {
            return false;
        }
        $period = $this->sprintPeriod();
        return !empty($period) ? $period->contains(Carbon::now()) : false;
    }
```
Then removed the weekday check in the SprintChartService.